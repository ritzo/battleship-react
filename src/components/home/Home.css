.home {
    padding-left: 20px;
}

.play {
    margin: 0 50px;
    padding: 10px;
}
