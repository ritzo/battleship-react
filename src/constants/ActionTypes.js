// Actions related to the game setup
export const SETUP_BOARD = 'SETUP_BOARD';
export const SET_NAME = 'SET_NAME';
export const ADD_SHIP = 'ADD_SHIP';
export const PLAY = 'PLAY';
// Actions related to the game itself
export const SURRENDER = 'SURRENDER';
export const SHOOT = 'SHOOT';
export const OPONENT_TURN = 'OPONENT_TURN';
