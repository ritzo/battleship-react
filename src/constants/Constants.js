export const STATES = {
  OPEN: 'open',
  PLAYING: 'playing',
  FINISHED: 'finished',
};

export const SHIPS = [2, 3, 3, 3, 4];

export const POSITIONS = {
  HORIZONTAL: 'horizontal',
  VERTICAL: 'vertical',
};

export const PLAYERS = {
  PLAYER: 'player',
  OPONENT: 'cpu',
};

export const SQUARE_STATES = {
  EMPTY: 'empty',
  INVALID: 'invalid',
  SHIP: 'ship',
  SHOOT_MISS: 'shoot_miss',
  SHOOT_SHIP: 'shoot_ship',
};

export const BOARD_TYPES = {
  EDITION: 'edition',
  PLAYER: 'player',
  OPONENT: 'oponent',
};
