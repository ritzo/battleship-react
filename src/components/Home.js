import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import BoardEditorContainer from '../containers/BoardEditorContainer';
import { BOARD_TYPES } from '../constants/Constants';

import './Home.css';

class Home extends React.Component {
  constructor(props) {
    super(props);

    this.gameData = props.gameData;
    this.actions = props.actions;

    this.name = this.gameData.player;
    this.playerBoard = this.gameData.playerBoard;
  }

  handleChange(event) {
    this.name = event.target.value;
    this.actions.setName(this.name);
  }

  render() {
    return (
      <div>
        <h1>Home</h1>

        <span>Name:</span>
        <input type="text" value={this.name} onChange={event => this.handleChange(event)} />

        <BoardEditorContainer
          title="Place your ships admiral"
          type={BOARD_TYPES.EDITION}
          squares={this.playerBoard}
          clickAction={this.clickAction}
        />

        <Link to="/game">Play</Link>
      </div>
    );
  }
}

Home.propTypes = {
  gameData: PropTypes.shape({
    player: PropTypes.string,
    playerBoard: PropTypes.array.isRequired,
    availableShips: PropTypes.array.isRequired,
    state: PropTypes.string.isRequired,
  }).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  actions: PropTypes.object.isRequired,
};

export default Home;
