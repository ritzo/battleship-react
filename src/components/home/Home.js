import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';

import BoardEditorContainer from '../board/BoardEditorContainer';
import { BOARD_TYPES, STATES } from '../../constants/Constants';

import './Home.css';

class Home extends React.Component {
  constructor(props) {
    super(props);

    this.actions = props.actions;
  }

  handleChange(event) {
    this.actions.setName(event.target.value);
  }

  handlePlayClick() {
    this.actions.play();
  }

  render() {
    const {
      player,
      availableShips,
      state,
    } = this.props;

    if (state === STATES.FINISHED) {
      return <Redirect to="/gameOver" />;
    }

    if (state === STATES.PLAYING) {
      return <Redirect to="/game" />;
    }

    return (
      <div className="home">
        <h1>Home</h1>

        <span>Name:</span>
        <input type="text" value={player} onChange={event => this.handleChange(event)} />

        <BoardEditorContainer
          title="Place your ships admiral"
          type={BOARD_TYPES.EDITION}
        />

        <button type="button" className="play" onClick={() => this.handlePlayClick()} disabled={availableShips.length}>Play</button>
      </div>
    );
  }
}

Home.propTypes = {
  player: PropTypes.string.isRequired,
  availableShips: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
  state: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    play: PropTypes.func.isRequired,
  }).isRequired,
};

export default Home;
