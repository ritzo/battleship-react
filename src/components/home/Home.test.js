import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';

import { STATES } from '../../constants/Constants';

describe('Home', () => {
  it('Snapshot', () => {
    const player = 'someone';
    const availableShips = [2, 3];
    const state = STATES.PLAYING;
    const actions = {
      play() {},
    };

    const enzymeWrapper = shallow(
      <Home player={player} availableShips={availableShips} state={state} actions={actions} />,
    );

    expect(enzymeWrapper).toMatchSnapshot();
  });
});
