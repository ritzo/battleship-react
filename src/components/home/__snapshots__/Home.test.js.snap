// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot 1`] = `
<Redirect
  push={false}
  to="/game"
/>
`;
